core/node/auth/auth_impl.go

```diff
@@ -779,9 +800,53 @@ func (ca *chainAuth) checkEntitlement(
 		}
 	}
 
+	// Look for any returned errors. If at least one check was positive, then we ignore any subsequent
+	// errors. Otherwise we will report an error result since we could not conclusively determine that
+	// the user was not a space member.
+	for err := range isMemberError {
+		// Once we encounter a positive entitlement result, we cancel all other request, which should result
+		// in context cancellation errors being returned for those checks, even though the check itself was
+		// not faulty. However, a context cancellation error can also occur if a server request times out, so
+		// not all cancellations can be ignored.
+		// Here, we collect all errors and report them, assuming that when the isMember result is false,
+		// no contexts were cancelled by us and therefore any errors that occur at all are informative.
+		if err != nil {
+			if membershipError != nil {
+				membershipError = fmt.Errorf("%w; %w", membershipError, err)
+			} else {
+				membershipError = err
+			}
+		}
+	}
+
 	if !isMember {
-		log.Warn("User is not a member of the space", "userId", args.principal, "spaceId", args.spaceId)
-		return &boolCacheResult{allowed: false}, nil
+		if membershipError != nil {
+			log.Error(
+				"User membership could not be evaluated",
+				"userId",
+				args.principal,
+				"spaceId",
+				args.spaceId,
+				"wallets",
+				wallets,
+				"aggregateError",
+				membershipError,
+			)
+			return &boolCacheResult{allowed: false}, membershipError
```

Comment:
why not to return nil value in case of the error? would it lead to more robust code when incorrect result can't be cached?